Gems/ROS2/Code/Source/Clock/SimulationClock.cpp

```diff
@@ -73,4 +73,10 @@ namespace ROS2
         AZStd::sort(frameTimeSorted.begin(), frameTimeSorted.end());
         m_currentMedian = frameTimeSorted[frameTimeSorted.size() / 2];
     }
+
+    void SimulationClock::ResetPublisher()
```

Comment:
Could you please describe (document) why only reseting the publisher is preferred (keeping the state of m_currentMedian and m_frameTimes? Why not just reset the clock object pointer?

Reply:
Resetting the clock object pointer is better. There isn't a reason to keep these states between simulation runs. I'll update the code.